### PR TITLE
Change the build status to Unstable when Jira Issues not DONE and deploying to D

### PR DESCRIPTION
**Pull request by jorge-romero**
Change to unstable the result of the build in case of documents not closed

### CODE REVIEW

File: vars/odsOrchestrationPipeline.groovy [hunk 1]
```diff
@@ -82,7 +84,17 @@ def call(Map config) {
...
+                def cannotContinueAsHasOpenIssuesInClosingRelease = false
+                try {
+                    result = new InitStage(this, project, repos, startAgentStage).execute()
+                } catch (OpenIssuesException ex) {
+                    cannotContinueAsHasOpenIssuesInClosingRelease = true
+                }
+                if (cannotContinueAsHasOpenIssuesInClosingRelease) {
+                    logger.warn('Cannot continue as it has open issues in the release.')
+                    return
+                }
```
Comment (metmajer) [inline vars/odsOrchestrationPipeline.groovy:97]: @jorge-romero why rather complex? wouldn't the logger.warn + return inside the catch block be enough?

File: src/org/ods/orchestration/util/Project.groovy [hunk 2]
```diff
@@ -362,20 +363,10 @@ class Project {
...
-                'since the following issues are work in progress: ':
-                "The pipeline failed since the following issues are work in progress (no documents were generated): "
-
-            this.getWipJiraIssues().each { type, keys ->
-                def values = keys instanceof Map ? keys.values().flatten() : keys
-                if (!values.isEmpty()) {
-                    message += '\n\n' + type.capitalize() + ': ' + values.join(', ')
-                }
-            }
+            String message = generateWIPIssuesMessage()
```
Comment (angelmp01) [inline src/org/ods/orchestration/util/Project.groovy:366]: If you use only the message inside the if, create it inside

File: src/org/ods/orchestration/util/Project.groovy [hunk 3]
```diff
@@ -406,6 +397,21 @@ class Project {
         return this
     }
 
+    def generateWIPIssuesMessage() {
```
Comment (angelmp01) [inline src/org/ods/orchestration/util/Project.groovy:400]: Maybe we can move the responsability of generate this message to another class instead Project class. Think that project class is too big.

File: vars/odsOrchestrationPipeline.groovy (hunk 1 above)
Comment (jorge-romero) [reply vars/odsOrchestrationPipeline.groovy:97]: That's was my first implementation, but it was complaining about trying to finalize the pipeline within a catch block, that's why I had to rewrite in this ugly way unfortunally

File: src/org/ods/orchestration/util/Project.groovy (hunk 2 above)
Comment (jorge-romero) [reply src/org/ods/orchestration/util/Project.groovy:366]: If you check the whole code as in the diff looks different it is inside the if block :)

File: src/org/ods/orchestration/util/Project.groovy (hunk 3 above)
Comment (jorge-romero) [reply src/org/ods/orchestration/util/Project.groovy]: Moved to another class, and moved the tests for that specific functionallity to a new Spec for that class

File: vars/odsOrchestrationPipeline.groovy (hunk 1 above)
Comment (angelmp01) [reply vars/odsOrchestrationPipeline.groovy:97]: Another solution may be logger.warn + result=false inside the catch block. And remove cannotContinueAsHasOpenIssuesInClosingRelease var.

File: src/org/ods/orchestration/util/Project.groovy [hunk 4]
```diff
@@ -406,6 +397,21 @@ class Project {
...
+            'since the following issues are work in progress: ' :
+            "The pipeline failed since the following issues are work in progress (no documents were generated): "
+
+        this.getWipJiraIssues().each { type, keys ->
+            def values = keys instanceof Map ? keys.values().flatten() : keys
+            if (!values.isEmpty()) {
+                message += '\n\n' + type.capitalize() + ': ' + values.join(', ')
+            }
+        }
+        message
```
[the remark this answers is not in the document]
Comment (jorge-romero) [reply src/org/ods/orchestration/util/Project.groovy]: It is done in the new class afte refactoring that we have created to deal with the message  as @angelmp01 requested

**Review by angelmp01 — APPROVED** [review]

**Pull request merged**